### ISSUE TITLE
Parse .gitignore files

### ISSUE THREAD

**Issue by gergelypap** [klausi/pareviewsh#7]
**Problem**

On line https://github.com/klausi/pareviewsh/blob/7.x-1.x/pareview.sh#L164, you check if some folders exist, and fail if any of them found.

Lots of people have `node_modules` and `vendor` in their projects which are almost always gitignored.

**Proposal**

I think you should parse any `.gitignore` (or other VCS ignore) files in the working directory of pareview.sh, so we don't need to remove the whole node_modules folder in order for pareview to run as expected.

And if a vendor/system folder or file is missing from .gitignore, but present in the project, pareview should then throw an ERROR.

What do you think?

**Comment by klausi** [comment]
vendor or node_modules folders are not allowed in drupal.org projects, so I think it is not worth to spend the effort to make special exceptions for them.

We could exclude those folder from the find commands that gather files to check and instead add an error that vendor or node_modules are not allowed to be committed on drupal.org.

**Comment by gergelypap** [comment]
Exactly, but this is what .gitignore is for, isn't it? Ignored folders _will not be pushed_ to drupal.org repositories. (Because all drupal.org projects are required to be pushed to Drupal's own Git system, http://cgit.drupalcode.org/)

I've decided to use SASS in my [contrib module](https://www.drupal.org/project/parade), and [Gulp](http://gulpjs.com/) for watching file changes and automatically ESLint + CSSComb the results.

Of course people can still push "node_modules", but I think pareview should throw an error when:
1. node_modules, vendor, etc... folders are present,
2. AND they are not listed in the .gitignore of the project.
